### PR TITLE
Add reference sections and implement new childpages shortcode

### PR DESCRIPTION
**Pull request by nadinequinr3**
- change `corda-5` in front matter to `corda5` -> Hugo doesn't like the "-"
- copy over existing references material (APIs, config fields, CLI commands- previously reviewed)
- utilise new childpages shortcode
- update links on the homepage

Preview here: https://docs.preview.docs.r3.com/

### CODE REVIEW

File: content/en/platform/corda/5.0/developing-applications/api/application/application.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  corda5:
+    identifier: corda5-api-application
+    parent: corda5-api
+    weight: 1000
+section_menu: corda5
+---
+
+The `corda-application` module provides the fundamental building blocks required to create a [flow](../../../introduction/key-concepts.html#flows) and so all CorDapps use this module.
+
+`corda-application` sits at a higher level in the module hiarachy and exposes the following modules as API dependencies:
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/developing-applications/api/application/application.md:14]:
```suggestion
`corda-application` sits at a higher level in the module hierachy and exposes the following modules as API dependencies:
```

File: content/en/platform/corda/5.0/developing-applications/api/application/persistence.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+---
+
+The `persistence` package provides services for performing persistence operations; mainly reading and writing data to and from the database. The `PersistenceService` is the main service for providing this functionality.
+
+Corda 5 supports CRUD (Create, Read, Update, Delete) operations for user-defined types. This is achieved using JPA-annotated entities and, to manage database migrations, Liquibase.
+
+## Defining Custom Tables Using Liquibase Migrations
+
+[CorDapps](../../../introduction/key-concepts.html#cordapps) store data in a relational database.
+When Corda creates a [virtual node](../../../introduction/key-concepts.html#virtual-nodes) for a CorDapp (as part of a [CPI](../../../introduction/key-concepts.html#corda-package-installer-cpi)), it requires associated tables, indexes, foreign-keys, etc.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/developing-applications/api/application/persistence.md:19]:
```suggestion
When Corda creates a [virtual node](../../../introduction/key-concepts.html#virtual-nodes) for a CorDapp (as part of a [CPI](../../../introduction/key-concepts.html#corda-package-installer-cpi)), it requires associated tables, indexes, foreign-keys, and so on.
```

File: content/en/platform/corda/5.0/developing-applications/api/application/persistence.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+To create these, you must embed Liquibase files into the CorDapp [CPK](../../../introduction/key-concepts.html#corda-packages-cpks).
+
+Liquibase manages DB changes in a “Change Log” which references one or more change sets.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/developing-applications/api/application/persistence.md:22]:
```suggestion
Liquibase manages database changes in a “Change Log” which references one or more change sets.
```

File: content/en/platform/corda/5.0/developing-applications/api/application/persistence.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+You must specify the top level `databaseChangeLog` in a resource file in the CPK called `migration/db.changelog-master.xml`.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/developing-applications/api/application/persistence.md:23]:
```suggestion
You must specify the top-level `databaseChangeLog` in a resource file in the CPK called `migration/db.changelog-master.xml`.
```

File: content/en/platform/corda/5.0/developing-applications/api/application/serialization.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+title: "net.corda.v5.application.serialization"
+menu:
+  corda5:
+    identifier: corda5-api-app-serialization
+    parent: corda5-api-application
+    weight: 7000
+section_menu: corda5
+---
+
+The `serialization` package provides services for working with data marked as `@CordaSerializable`, in order to render it into a form suitable for sending to a counterparty. Usually this is handled by the messaging layer (see the <a href="messaging.md">`messaging` package</a>), but if you require access to serialization directly it is exposed here. The main service for serailization is the <a href="../../../../../../api-ref/corda/5.0-beta/java/net/corda/v5/application/serialization/SerializationService.html" target="_blank">`SerializationService`</a>. At present, the only scheme available via this service is AMQP.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/developing-applications/api/application/serialization.md:12]:
```suggestion
The `serialization` package provides services for working with data marked as `@CordaSerializable`, in order to render it into a form suitable for sending to a counterparty. Usually this is handled by the messaging layer (see the <a href="messaging.md">`messaging` package</a>), but if you require access to serialization directly it is exposed here. The main service for serialization is the <a href="../../../../../../api-ref/corda/5.0-beta/java/net/corda/v5/application/serialization/SerializationService.html" target="_blank">`SerializationService`</a>. At present, the only scheme available via this service is AMQP.
```

File: content/en/platform/corda/5.0/reference/corda-cli/database.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+menu:
+  corda5:
+    identifier: corda5-cordacli-database
+    weight: 1000
+    parent: corda5-cli-reference
+section_menu: corda5
+title: "database"
+---
+
+This section lists the Corda CLI `database` arguments. You can use these commands to manually perform setup actions in the database, as described in the <!--[Manual Bootstrapping Tutorial]() section-->.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/database.md]:
```suggestion
This section lists the Corda CLI `database` arguments. You can use these commands to manually perform setup actions in the database, as described in the <!--[Manual Bootstrapping Tutorial]() section-->.
```
link missing

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+menu:
+  corda5:
+    identifier: corda5-cordacli-initial-config
+    weight: 2000
+    parent: corda5-cli-reference
+section_menu: corda5
+title: "initial-config"
+---
+
+This section lists the Corda CLI `initial-config` arguments. You can use these commands to manually perform various setup actions, <!--as described in the [Manual Bootstrapping Tutorial]().-->
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md]:
```suggestion
This section lists the Corda CLI `initial-config` arguments. You can use these commands to manually perform various setup actions, <!--as described in the [Manual Bootstrapping Tutorial]().-->
```
link missing

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+{{% tab name="Windows" %}}
+```shell
+corda-cli.cmd initial-config create-user-config -u <INITIAL-ADMIN-USERNAME> -p <INITIAL-ADMIN-PASSWORD> -l /tmp/db
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+## create-db-config
+
+The `create-db-config` command creates the SQL statements to insert the connection manager configuration for the database.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md:44]:
```suggestion
The `create-db-config` command creates the SQL statements to insert into the connection manager configuration for the database.
```
Guessing

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+| <div style="width:160px">Argument</div> | Description                                                                                                                                      |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| -a, \-\-is-admin                        | Specifies if this is an admin (DDL) connection. The default value is false.                                                                      |
+| -d, \-\-description                     | Detailed information about the database connection.                                                                                              |
+| -e, \-\-passphrase                      | The passphrase for the encrypting secrets service.  This must match the value specified in the Corda deployment configuration for the DB worker. |
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md:50]:
```suggestion
| -e, \-\-passphrase                      | The passphrase for the encrypting secrets service.  This must match the value specified in the Corda deployment configuration for the database worker. |
```

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+| -j, \-\-jbdc-url                        | The JDBC URL for the connection. This value is required.                                                                                         |
+| \-\-jdbc-pool-max-size                  | The maximum size of the JDBC connection pool. The default value is 10.                                                                           |
+| -l, \-\-location                        | The path to write the generated SQL files to.                                                                                                    |
+| -n, \-\-name                            | The name of the database connection. Required.                                                                                                   |
+| -p, \-\-password                        | The password for the database connection. Required.                                                                                              |
+| -s, \-\-salt                            | Salt for the encrypting secrets service. This must match the value specified in the Corda deployment configuration for the DB worker.            |
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md:56]:
```suggestion
| -s, \-\-salt                            | Salt for the encrypting secrets service. This must match the value specified in the Corda deployment configuration for the database worker.            |
```

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 11]
```diff
@@ -0,0 +1,111 @@
...
+{{< /tabs >}}
+
+## create-crypto-config
+
+The `create-crypto-config` command creates the SQL statements to insert the initial crypto configuration for the database. This operation must be performed after the cluster database is initialized but before the cluster is started.
+
+| <div style="width:160px">Argument</div> | Description                                                                                                                                      |
+| --------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
+| -l, \-\-location                        | The path to write the generated SQL files to.                                                                                                    |
+| -p, \-\-passphrase                      | The passphrase for the encrypting secrets service.  This must match the value specified in the Corda deployment configuration for the DB worker. |
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md:90]:
```suggestion
| -p, \-\-passphrase                      | The passphrase for the encrypting secrets service.  This must match the value specified in the Corda deployment configuration for the database worker. |
```

File: content/en/platform/corda/5.0/reference/corda-cli/initial-config.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+| -s, \-\-salt                            | Salt for the encrypting secrets service. This must match the value specified in the Corda deployment configuration for the DB worker.            |
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/initial-config.md:91]:
```suggestion
| -s, \-\-salt                            | Salt for the encrypting secrets service. This must match the value specified in the Corda deployment configuration for the database worker.            |
```

File: content/en/platform/corda/5.0/reference/corda-cli/package.md [hunk 13]
```diff
@@ -0,0 +1,130 @@
...
+menu:
+  corda5:
+    identifier: corda5-cordacli-develop-commands
+    weight: 4000
+    parent: corda5-cli-reference
+section_menu: corda5
+title: "package"
+---
+
+This section lists the Corda CLI `package` arguments. You can use these commands to execute operations for working with CPB and CPI files, as described in the <!--[CorDapp Packaging]() tutorial-->.
```
Comment (AnnaK-R3) [inline content/en/platform/corda/5.0/reference/corda-cli/package.md]:
```suggestion
This section lists the Corda CLI `package` arguments. You can use these commands to execute operations for working with CPB and CPI files, as described in the <!--[CorDapp Packaging]() tutorial-->.
```
link missing

File: content/en/platform/corda/5.0/reference/_index.md [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+version: 'Corda 5.0'
+date: 2023-04-24
+menu:
+  corda5:
+    identifier: corda5-references
+    weight: 6000
+section_menu: corda5
+---
+This section contains reference material for use by multiple Corda personas. It contains the following:
+* [Configuration Fields]({{< relref "./config-fields/_index.md" >}})
```
Comment (yarivus) [inline content/en/platform/corda/5.0/reference/_index.md:13]: why not use {{childpages}} here too ?

**Review by yarivus — COMMENTED** [review]
I see pending comments from Anna

File: content/en/platform/corda/5.0/reference/_index.md (hunk 14 above)
Comment (nadinequinr3) [reply content/en/platform/corda/5.0/reference/_index.md:13]: The REST API page is a HTML file that's generated from the code. It doesn't contain Hugo front matter to use to sort the list of child pages.

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — COMMENTED** [review]
thx @nadinequinr3 and @AnnaK-R3

**Pull request merged**